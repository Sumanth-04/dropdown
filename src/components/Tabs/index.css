.tab_ul {
    list-style: none;
}

.tab_ul > li{
    display: inline;
    list-style-type: none;
    padding-right: 20px;
}

.tab-list > li::after{
    background: none repeat scroll 0 0 transparent;
    bottom: 0;
    content: "";
    display: block;
    height: 2px;
    left: 50%;
    position: absolute;
    background: #fff;
    transition: width 0.3s ease 0s, left 0.3s ease 0s;
    width: 0;
}

.tab-list > li :hover ::after { 
    content: "";
    width: 100%; 
    left: 0; 
}

.rd_tabs { 
    padding: 0; 
    list-style: none; 
}

.rd-tab { 
    display: table-cell; 
    position: relative;
}

.rd-tab-item {
    color: #333;
    text-transform: uppercase;
    text-decoration: none;
    letter-spacing: 0.15em;
    
    display: inline-block;
    padding: 15px 20px;
    position: relative;
  }

.rd-tab-item:after{
    background: none repeat scroll 0 0 transparent;
    bottom: 0;
    content: "";
    display: block;
    height: 2px;
    position: absolute;
    background: #333;
    transition: width 0.3s ease 0s, left 0.3s ease 0s;
    width: 0;
  }

.rd-tab-item:hover:after,
.rd-tab-item:focus:after,
.rd-tab-item:active:after { 
    width: 100%;
    left: 0; 
}