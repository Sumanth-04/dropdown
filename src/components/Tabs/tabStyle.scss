@import '../variables';
//tabs

.rd_tabs {
  width: 100%;
  margin-bottom: 8px;
  // border-bottom: 1px solid #EDEDED;
}

.rd-tab {
  .tab-list-active {
    font-family: ChivoRegular !important;
  }
  .rd-tab-item {
    font-size: $tabLinkFontSize;
    font-family: ChivoLight;
    padding: 0px;
    margin-right: 24px;
    letter-spacing: 1px;
    
    &:hover{
      text-decoration: none;
      color: black;
    } 
  }
}

// pills

.rd_pills {
  width: 100%;
}

.rd_pill {
  width: 100%;
  margin-bottom: 8px;
  border-bottom: none !important;
}
.rd-pill {
  display: inline;
  position: relative;
  margin-right: 1.2rem;
  .pill-list-active {
    font-family: ChivoLight !important;
    background: $primary-color;
    color: white;
    &:hover{
      text-decoration: none;
      color: white !important;
      cursor: pointer;
    } 
  }
  .rd-pill-item {
    font-size: $tabPillFontSize;
    font-family: ChivoLight;
    text-transform: uppercase;
    padding: 5px 10px;
    letter-spacing: 1px;
    border: 1px solid $primary-color;
    &:hover{
      text-decoration: none;
      color: black;
      cursor: pointer;
    } 
  }
}