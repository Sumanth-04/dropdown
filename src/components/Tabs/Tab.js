import React, { Component } from 'react';
import PropTypes from 'prop-types';
import './index.css'
import './tabStyle.scss'

class Tab extends Component {
  static propTypes = {
    activeTab: PropTypes.string.isRequired,
    label: PropTypes.string.isRequired,
    onClick: PropTypes.func.isRequired,
  };

  onClick = () => {
    const { label, onClick } = this.props;
    onClick(label);
  }

  render() {
    const {
      onClick,
      props: {
        activeTab,
        label,
        action
      },
    } = this;

    let className = `${this.props.NavType}-list-item`;

    if (activeTab === label) {
      className += ` ${this.props.NavType}-list-active`;
    }

    return (
            <li className={`rd-${this.props.NavType}`} onClick={action}>
              <a className={`rd-${this.props.NavType}-item ${className}`}  onClick={onClick}>{label}</a>
            </li>
    );
  }
}

export default Tab;