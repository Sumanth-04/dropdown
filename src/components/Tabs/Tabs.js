import React, { Component } from 'react';
import Tab from './Tab'
import PropTypes from 'prop-types';
import './index.css'
import './tabStyle.scss'

class Tabs extends Component {
	static propTypes = {
		onClick: PropTypes.func,
		children: PropTypes.instanceOf(Array).isRequired
	}

	constructor(props) {
		super(props);

		this.state = {
			activeTab: this.props.children[0].props.label,
		};
	}

	onClickTabItem = (tab) => {
		this.setState({ activeTab: tab });
	}

	render() {
		//console.log(this.props.children)
		const {
			onClickTabItem,
			props: {
				children,
				NavType
			},
			state: {
				activeTab,
			}
		} = this;

		return (
			<div>
				<ul className={`rd_${NavType}s`}>
					{children.map((child) => {
						const { label, action } = child.props;
						return (
							<Tab
								NavType={NavType}
								activeTab={activeTab}
								key={label}
								label={label}
								onClick={onClickTabItem}
								action={action}
							/>
						);
					})}
				</ul>
				<div className="tab-content">
					{children.map((child) => {
						if (child.props.label !== activeTab) return undefined;
						return child.props.children;
					})}
				</div>
			</div>
		)
	}
}

Tabs.defaultProps = {
	NavType: 'tab'
};
  
export default Tabs;