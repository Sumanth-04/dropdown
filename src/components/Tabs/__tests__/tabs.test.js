import React from 'react';
import renderer from 'react-test-renderer';
import Tabs from '../index'

it('Tab Renders with Content', () => {
	const tree = renderer
		.create(
			<Tabs>
				<div label="1">Hi</div>
				<div activeTab='2' label="2">Sup</div>
				<div label="3">Hello</div>
			</Tabs>
		)
		.toJSON();
	expect(tree).toMatchSnapshot();
});