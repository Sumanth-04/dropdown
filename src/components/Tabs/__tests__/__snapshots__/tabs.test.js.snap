// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tab Renders with Content 1`] = `
<div>
  <ul
    className="rd_tabs"
  >
    <li
      className="rd-tab"
    >
      <a
        className="rd-tab-item tab-list-item tab-list-active"
        onClick={[Function]}
      >
        1
      </a>
    </li>
    <li
      className="rd-tab"
    >
      <a
        className="rd-tab-item tab-list-item"
        onClick={[Function]}
      >
        2
      </a>
    </li>
    <li
      className="rd-tab"
    >
      <a
        className="rd-tab-item tab-list-item"
        onClick={[Function]}
      >
        3
      </a>
    </li>
  </ul>
  <div
    className="tab-content"
  >
    Hi
  </div>
</div>
`;

exports[`Tab Renders with Content 2`] = `
<div>
  <ul
    className="rd_tabs"
  >
    <li
      className="rd-tab"
    >
      <a
        className="rd-tab-item tab-list-item tab-list-active"
        onClick={[Function]}
      >
        1
      </a>
    </li>
    <li
      className="rd-tab"
    >
      <a
        className="rd-tab-item tab-list-item"
        onClick={[Function]}
      >
        2
      </a>
    </li>
    <li
      className="rd-tab"
    >
      <a
        className="rd-tab-item tab-list-item"
        onClick={[Function]}
      >
        3
      </a>
    </li>
  </ul>
  <div
    className="tab-content"
  >
    Hi
  </div>
</div>
`;
